src/main/java/com/iridium/iridiumskyblock/managers/SchematicManager.java

```diff
@@ -115,4 +125,78 @@ private CompletableFuture<Void> pasteSchematic(Island island, Schematics.Schemat
         return completableFuture;
     }
 
+    public boolean processTransaction(Player player, Schematics.SchematicConfig schematic) {
+
+        User user = IridiumSkyblock.getInstance().getUserManager().getUser(player);
+        Optional<Island> island = IridiumSkyblock.getInstance().getTeamManager().getTeamViaID(user.getTeamID());
+
+        if(!island.isPresent()){
+            //NO ISLAND
+            return false;
+        }
+
+        TeamBank bank;
+
+        double bankBalance;
+        List<Placeholder> bankPlaceholders;
+        double moneyCost = schematic.regenCost.money;
+        Economy economy = IridiumSkyblock.getInstance().getEconomy();
+
+        if(moneyCost == 0 || economy != null && economy.getBalance(player) >= moneyCost) {
+            //INVALID TRANSACTION
+            return false;
+        }
+
+        for (String schematicBankItem : schematic.regenCost.bankItems.keySet()) {
+
+            bank = IridiumSkyblock.getInstance().getTeamManager().getTeamBank(island.get(), schematicBankItem);
+            bankBalance = bank.getNumber();
+
+            double cost = round(schematic.regenCost.bankItems.get(schematicBankItem), 2);
+
+            if (bank.getNumber() < cost) {
+                player.sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().cannotAfford
+                        .replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)));
+                IridiumSkyblock.getInstance().getSchematics().failSound.play(player);
+
+                return false;
+            }
+
+            double finalBankBalance = bankBalance;
+            bankPlaceholders = IridiumSkyblock.getInstance().getBankItemList().stream()
+                    .map(BankItem::getName)
+                    .map(name -> new Placeholder(name + "_cost", formatPrice(finalBankBalance)))
+                    .collect(Collectors.toList());
+
+            player.sendMessage(StringUtils.color(StringUtils.processMultiplePlaceholders(IridiumSkyblock.getInstance().getMessages().paidForRegen
+                            .replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)
+                            .replace("%player%", player.getName())
+                            .replace("%schematic%", StringUtils.color(schematic.item.displayName))
+                            .replace("%bank_cost%", formatPrice(cost))
+                            .replace ("%vault_cost%", formatPrice(moneyCost)),
+                    bankPlaceholders)
+            ));
```

Comment:
This also needs to be outside the for loop otherwise we will send the player a message for each BankItem

Reply:
yeah, i wanted to add bank items to the message, tho im not sure how to add each and every bank item and then just the one vault eco without separating the message (rn they cant see they're paying both)